src/memory_server.ml

```diff
@@ -45,9 +45,9 @@ let wrap dbg f =
 		(* NB both needed and free have been inflated by the lowmem_emergency_pool etc *)
 		let needed = Int64.sub needed Squeeze_xen.target_host_free_mem_kib
 		and free = Int64.sub free Squeeze_xen.target_host_free_mem_kib in
-		raise (Memory_interface.Cannot_free_this_much_memory (needed, free))
+		raise (MemoryError (Memory_interface.Cannot_free_this_much_memory (needed, free)))
```

Comment:
Is there a way to return the error and use the `err` construct to automatically decide if we are raising errors or returning `Error`s?